docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx

```diff
@@ -256,6 +253,58 @@ height={593}
 
 ---
 
+## Arbitary Python AutomationConditions <Experimental />
+
+Some automation use cases require custom business logic that cannot be expressed with off-the-shelf components. In these cases, you can define AutomationConditions which execute arbitrary python code, and compose them with the built-in conditions.
+
+### Setup
+
+By default, Dagster executes `AutomationConditionSensorDefinitions` in a daemon process that does not have access to your user code. In order to execute arbitrary Python code, you'll need to update this to execute on your user code server. This is the same place that your `@sensor` methods are evaluated.
+
+<Note>
+  Automation condition evaluation can be more resource-intensive than a typical
+  sensor. A limit of 500 assets or checks per sensor is enforced.
+</Note>
+
+To do this, add an automation condition sensor to your definitions with the `use_user_code_server` flag set to `True`:
+
+```python
+import dagster as dg
+
+defs = dg.Definitions(
+  sensors=[dg.AutomationConditionSensorDefinition("automation_condition_sensor", target=dg.AssetSelection.all(), use_user_code_server=True)]
+)
+```
+
+This will allow your sensor to target automation conditions containing custom python code.
+
+### Defining a custom condition
+
+You can create your own subclass of `AutomationCondition`, defining the `evaluate()` method. For example, imagine you want to avoid executing anything on a company holiday. To do this, you can first define a condition which detects if it's currently a company holiday:
+
+```python
+import dagster as dg
+
+class IsCompanyHoliday(dg.AutomationCondition):
+  def evaluate(self, context: dg.AutomationContext) -> dg.AutomationResult:
+    if is_company_holiday(context.evaluation_time):
+      true_subset = context.candidate_subset
+    else:
+      true_subset = context.get_empty_subset()
+    return dg.AutomationResult(true_subset, context=context)
```

Comment:
Some comments explaining the naming convention here would be illuminating to an interested observer. Explain what a candidate subset is, for example.